distributed/scheduler.py

```diff
@@ -7736,7 +7737,10 @@ def check_idle(self) -> int | None:
             return self.idle_since
 
         if self.idle_timeout:
-            if time() > self.idle_since + self.idle_timeout:
+            if (
+                time() > self.idle_since + self.idle_timeout
+                and self.check_idle_jupyter()
```

Comment:
This will only get called if `self.idle_timeout` is set, but when calling in via the HTTP API this is not guaranteed.